src/components/UncontolledForm/UncontrolledForm.tsx

```diff
@@ -0,0 +1,212 @@
+import { useRef, useState } from 'react';
+import { IFormData, IFormDataStored } from '../../utils/types/types';
+import CountryAutocomplete from '../CountryAutocomplete/CountryAutocomplete';
+import { useAppDispatch } from '../../stores/hooks/redux';
+import { updateFormData } from '../../stores/reducers/formDataSlice';
+import { Link, useNavigate } from 'react-router-dom';
+import { SCHEMA } from '../../utils/types/yup/shema';
+import { ValidationError } from 'yup';
+
+const UncontrolledForm = () => {
+  const dispatch = useAppDispatch();
+  const formRef = useRef<HTMLFormElement>(null);
+  const [acceptTerms, setAcceptTerms] = useState(false);
+  const [fileList, setFileList] = useState<File[]>([]);
+  const [formErrors, setFormErrors] = useState<Record<string, string>>({});
+  const navigate = useNavigate();
+  let formDataObject: IFormData = {
+    name: '',
+    age: 0,
+    email: '',
+    password: '',
+    confirmPassword: '',
+    gender: '',
+    terms: false,
+    country: '',
+    picture: [],
+  };
```

Comment:
I don't think you really need initial object. You reassign this object ```handleSubmit``` method. Even you have all empty inputs you will receive it from the ref. So I don't see necessity of ```let formDataObject``` here.
You only use this for "country" value, so you can leave only "country"